scheduler/server/job_state.go

```diff
@@ -151,13 +171,18 @@ func (j *jobState) taskStarted(taskId string, tr *taskRunner) {
 		delete(j.Completed, taskId)
 	}
 	j.Running[taskId] = taskState
+
+	// add the task to the map of tasks by start time
+	startTimeSec := taskState.TimeStarted.Truncate(time.Second)
```

Comment:
We are always truncating to the nearest time.Second but this is not enforced anywhere (e.g. by some sort of wrapping mechanism). If we ever change or add code and forget to do this it will probably break things.

Reply:
it's only local to this function - the whole intent of this function is to step backwards in time 1 second at a time.  Only the initializer (ln 572) and time decrement (ln 584)  should be modifying the time value.  If we're doing something else to modify the time value, yes we would need to compute the time then 'wrap' with Truncate() (In essence Truncate() is the only wrapper we could have in common to all lines of code impacting the time index.)